presto-spi/src/main/java/com/facebook/presto/spi/ConnectorSplit.java

```diff
@@ -13,13 +13,28 @@
  */
 package com.facebook.presto.spi;
 
+import com.facebook.presto.spi.schedule.NodeSelectionStrategy;
+
 import java.util.List;
 
 public interface ConnectorSplit
 {
-    boolean isRemotelyAccessible();
+    /**
+     * Indicate the node affinity of a Split
+     * 1. HARD_AFFINITY: Split is NOT remotely accessible and has to be on specific nodes
+     * 2. SOFT_AFFINITY: Connector level cache
```

Comment:
connector split provides a list of preferred nodes for engine to pick from but not mandatory.